django/applications/catmaid/control/tree_util.py

```diff
@@ -217,11 +219,11 @@ def lazy_load_trees(skeleton_ids, node_properties):
             tree = DiGraph()
 
         fields = {k: v for k,v in zip(props, islice(t, 3, 3 + len(props)))}
-        tree.add_node(t[0], fields)
+        tree.add_node(t[0], fields) # type: ignore # mypy cannot prove tree will have a value by this point
```

Comment:
Might be worth staring at this a bit more to make sure there's not a bug here.

Reply:
The logic is kind of obfuscated there, I agree. It is fine though, because every treenode in `ts` has a skeleton ID (it is part of the query). Therefore, if there are results, the first iteration will find a valid skeleton ID in `t[2]`, which is different from `None`, the initial `skid` value. Because of this, the first iteration (and every one with a changing skeleton ID) will create a new `DiGraph` for the current skeleton ID and store it `tree`.